### PR TITLE
Move Machine Setup code to use Flux

### PR DESCRIPTION
**Pull request by jeffdm**
Last part of Kitematic that was missing in this refactor.

- Change machine setup code to Flux
- Add loading page when Kitematic starts (instead of the white screen)
- Node v4 compatibility

### CODE REVIEW

File: src/stores/SetupStore.js [hunk 1]
```diff
@@ -1,202 +1,22 @@
...
+    this.percent = 0;
+    this.error = null;
+  }
+
+  error ({error}) {
+    this.setState({error});
+  }
+
+  progress ({progress}) {
+    this.setState({error: null, progress})
```
Comment (FrenchBen) [inline src/stores/SetupStore.js]: missing semicolon

File: src/menutemplate.js [hunk 2]
```diff
@@ -6,8 +6,8 @@ var util = require('./utils/Util');
 var setupUtil = require('./utils/SetupUtil');
 var metrics = require('./utils/MetricsUtil');
 var machine = require('./utils/DockerMachineUtil');
+var docker = require('./utils/DockerUtil');
 var dialog = remote.require('dialog');
```
Comment (FrenchBen) [inline src/menutemplate.js]: Can we convert these to imports to match all other files?

File: src/utils/SetupUtil.js [hunk 3]
```diff
@@ -1,113 +1,111 @@
...
+
+  async setup () {
+    metrics.track('Started Setup');
+    while (true) {
+      try {
+        if (!util.isWindows() && !virtualBox.active()) {
+          await util.exec(setupUtil.macSudoCmd(util.escapePath('/Library/Application Support/VirtualBox/LaunchDaemons/VirtualBoxStartup.sh') + ' restart'));
+        }
+
+        let exists = await virtualBox.vmExists('default');
```
Comment (FrenchBen) [inline src/utils/SetupUtil.js]: can we use `machine.name()` instead of `'default'` ?

**Comment by jeffdm** [comment]
Thanks @FrenchBen ! Updated.

**Comment by FrenchBen** [comment]
LGTM

**Comment by jeffdm** [comment]
Thanks @FrenchBen !

**Pull request merged**